src/core/credentials/transport/tls/spiffe_utils.cc

```diff
@@ -156,4 +167,114 @@ absl::StatusOr<SpiffeId> SpiffeId::FromString(absl::string_view input) {
   return SpiffeId(trust_domain, absl::StrCat("/", path));
 }
 
+void SpiffeBundle::JsonPostLoad(const Json&, const JsonArgs&,
+                                ValidationErrors* errors) {
+  {
+    ValidationErrors::ScopedField field(errors, ".keys");
+    for (size_t i = 0; i < keys_.size(); i++) {
+      auto root = keys_[i].GetRoot();
+      if (!root.ok()) {
+        errors->AddError(absl::StrFormat("Cannot get root certificate: %s",
+                                         root.status().ToString()));
+      } else {
+        roots_.push_back(*root);
+      }
+    }
+  }
+}
+
+void SpiffeBundleKey::JsonPostLoad(const Json&, const JsonArgs&,
+                                   ValidationErrors* errors) {
+  {
+    ValidationErrors::ScopedField field(errors, ".use");
+    if (use_ != kAllowedUse) {
+      errors->AddError(
+          absl::StrFormat("got %s. Only supported value for use field is %s.",
```

Comment:
Suggest formatting this error message more readably:

```
absl::StrFormat("value must be \"%s\", got \"%s\"", kAllowedUse, use_)
```

Reply:
Done